components/places/ffi/src/lib.rs

```diff
@@ -281,25 +327,25 @@ pub unsafe extern "C" fn sync15_history_sync(
     error: &mut ExternError,
 ) {
     log::debug!("sync15_history_sync");
-    CONNECTIONS.call_with_result(error, handle, |conn| -> places::Result<_> {
-        // XXX - this is wrong - we kinda want this to be long-lived - the "Db"
-        // should own the store, but it's not part of the db.
-        let store = HistoryStore::new(conn);
-        let mut sync_ping = telemetry::SyncTelemetryPing::new();
-        let result = store.sync(
+    APIS.call_with_result(error, handle, |api| -> places::Result<_> {
+        // Note that api.sync returns a SyncPing which we drop on the floor.
+        // XXX - we should consider exposing it over the FFI while making the
+        // change to PlacesApi.
```

Comment:
I was going to think about doing this as part of the patch, but the telemetry requirements still aren't clear (specifically, it's not clear if we want to return the sync ping directly, or some other "container" type holding that ping.) So for now, I suggest just removing the XXX comment here.